metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py

```diff
@@ -322,6 +335,18 @@ def __init__(self, config: SQLCommonConfig, ctx: PipelineContext, platform: str)
                 cached_domains=[k for k in self.config.domain], graph=self.ctx.graph
             )
 
+        self.views_processed: Set[str] = set()
```

Comment:
Maintaining `views_processed` here , so that we have an option to emit lineage for remaining (failed) views as part of native source. For example postgres source already uses view dependency tables to get table level lineage. Once we support partial view definitions, we can follow logic - 
1. attempt table+column lineage here and update views_processed for successful extraction.
2. fallback to native lineage extraction for views that are not already processed (i.e. failed in lineage extraction).

Reply:
What do you mean by "partial view definitions"?

Reply:
If the view is created using SQL

`CREATE VIEW db1.array_struct_test_view as select * from db1.array_struct_test;`

Then the view definition as retrieved from database - and set as viewDefinition.viewLogic is like
```
select `array_struct_test`.`property_id`, `array_struct_test`.`service` from `db1`.`array_struct_test`
```

Observed this for postgres and presto-on-hive. This may also be the case for a few others.

Reply:
Let's create a helper that leaves SqlParsingResult with query_type=CREATE as-is, but for type=SELECT it injects the out_table into the result

Reply:
done